### PR TITLE
MA-9: Create Directions page

### PR DESCRIPTION
**Pull request by vitalii-petrun**
recreated Roman's branch

based on last dev

image
image
image

### CODE REVIEW

File: lib/presentation/pages/directions_page/directions_page.dart [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+    final yOffset = renderSliver.constraints.precedingScrollExtent +
+        (renderSliver.geometry?.scrollExtent ?? 0);
+    return yOffset;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      backgroundColor: Colors.white,
```
Comment (vovaklh) [inline lib/presentation/pages/directions_page/directions_page.dart:36]: I think we don't need this

File: lib/presentation/pages/directions_page/directions_page.dart [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            key: _headerKey,
+            delegate: Header(
+              headerYOffset: _getHeaderYOffset,
+              screenWidth: context.screenWidth,
+            ),
+            pinned: true,
+          ),
+          const SliverSizedBox(height: 70),
+          const SliverSizedBox(child: _PartnershipSection()),
+          const SliverSizedBox(child: _BenefitsSection()),
```
Comment (vovaklh) [inline lib/presentation/pages/directions_page/directions_page.dart:49]: Shoul be there any spacing?

File: lib/presentation/pages/directions_page/directions_page.dart [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+          const SliverSizedBox(child: _CardsSection()),
+          const SliverSizedBox(child: _ProfitSection()),
```
Comment (vovaklh) [inline lib/presentation/pages/directions_page/directions_page.dart:51]: Add spaces pls

File: lib/presentation/pages/directions_page/sections/benefits_section.dart [hunk 4]
```diff
@@ -0,0 +1,33 @@
+part of "../directions_page.dart";
+
+class _BenefitsSection extends StatelessWidget {
+  const _BenefitsSection({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsets.symmetric(vertical: 100),
```
Comment (vovaklh) [inline lib/presentation/pages/directions_page/sections/benefits_section.dart:9]: Delete vertical padding and add spaces in page. Also add horizontal padding

File: lib/presentation/pages/directions_page/sections/benefits_section.dart [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+            context.localizations.auroraUniverse,
+            style: context.text.directionsBenefitsTitle,
+          ),
+          const SizedBox(height: 20),
+          Text(
+            context.localizations.tradeAndEarnMeetAndChat,
+            style: context.text.directionsBenefitsDescription,
+            textAlign: TextAlign.center,
+          ),
+          const SizedBox(height: 400), //TODO: replace this void area with images
```
Comment (vovaklh) [inline lib/presentation/pages/directions_page/sections/benefits_section.dart]: Move TODO above build method

File: lib/presentation/pages/directions_page/sections/cards_section.dart [hunk 6]
```diff
@@ -0,0 +1,163 @@
+part of '../directions_page.dart';
+
+class _CardsSection extends StatelessWidget {
+  const _CardsSection({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      color: context.color.directionsCardsSectionBackground,
+      padding: const EdgeInsets.symmetric(vertical: 100, horizontal: 20),
```
Comment (vovaklh) [inline lib/presentation/pages/directions_page/sections/cards_section.dart:10]: Remove vertical padding

File: lib/presentation/pages/directions_page/sections/cards_section.dart [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+        ),
+      ),
+    );
+  }
+}
+
+class _InfoBlock extends StatelessWidget {
+  const _InfoBlock({Key? key}) : super(key: key);
+  @override
+  Widget build(BuildContext context) {
```
Comment (vovaklh) [inline lib/presentation/pages/directions_page/sections/cards_section.dart:31]: Add empty line

File: lib/presentation/pages/directions_page/sections/cards_section.dart [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        Text(
+          context.localizations.keyBusinessLines,
+          style: context.text.metaverseThatUnitesTitle,
+        ),
+        const SizedBox(height: 20),
+        Text(context.localizations.usingSeveralBusiness),
```
Comment (vovaklh) [inline lib/presentation/pages/directions_page/sections/cards_section.dart:40]: Should we use style here?

File: lib/presentation/pages/directions_page/sections/cards_section.dart [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+      constraints: const BoxConstraints(maxWidth: 550),
+      child: StaggeredGrid.count(
+        crossAxisCount: 2,
+        mainAxisSpacing: 20,
+        crossAxisSpacing: 20,
+        children: [
+          const StaggeredGridTile.extent(
+            crossAxisCellCount: 1,
+            mainAxisExtent: 40,
+            child: SizedBox.shrink(),
```
Comment (vovaklh) [inline lib/presentation/pages/directions_page/sections/cards_section.dart:61]: Try to replace just by SizedBox()

File: lib/presentation/pages/directions_page/sections/cards_section.dart [hunk 10]
```diff
@@ -0,0 +1,163 @@
...
+            child: _DirectionCard(
+              icon: Icons.shield_outlined,
+              title: context.localizations.technology,
+              subtitle: context.localizations.virtualRealityTechnologies,
+            ),
+          ),
+          const StaggeredGridTile.extent(
+            crossAxisCellCount: 1,
+            mainAxisExtent: 40,
+            child: SizedBox.shrink(),
```
Comment (vovaklh) [inline lib/presentation/pages/directions_page/sections/cards_section.dart:102]: Try to replace just by SizedBox()

File: lib/presentation/pages/directions_page/sections/partnership_section.dart [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+
+    return Container(
+      decoration: const BoxDecoration(
+        image: DecorationImage(
+          image: AssetImage(AppAssets.indigoBackground),
+          fit: BoxFit.fill,
+        ),
+      ),
+      alignment: Alignment.center,
+      padding: const EdgeInsets.symmetric(vertical: 100),
```
Comment (vovaklh) [inline lib/presentation/pages/directions_page/sections/partnership_section.dart:21]: Remove vertical padding and add horizontal

File: lib/presentation/pages/directions_page/sections/profit_section.dart [hunk 12]
```diff
@@ -0,0 +1,115 @@
+part of '../directions_page.dart';
+
+class _ProfitSection extends StatelessWidget {
+  const _ProfitSection({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      padding: const EdgeInsets.symmetric(vertical: 100),
```
Comment (vovaklh) [inline lib/presentation/pages/directions_page/sections/profit_section.dart:9]: Remove vertical padding and add horizontal

File: lib/presentation/pages/directions_page/sections/profit_section.dart [hunk 13]
```diff
@@ -0,0 +1,115 @@
...
+        child: Column(
+          children: [
+            Text(
+              context.localizations.profitWithAurora,
+              style: context.text.blogWelcomeHeadline,
+              textAlign: TextAlign.center,
+            ),
+            const SizedBox(height: 20),
+            ConstrainedBox(
+              constraints: const BoxConstraints(maxWidth: 400),
```
Comment (vovaklh) [inline lib/presentation/pages/directions_page/sections/profit_section.dart:25]: I think you can replace it just by SizedBox

File: lib/presentation/pages/directions_page/sections/profit_section.dart [hunk 14]
```diff
@@ -0,0 +1,115 @@
...
+
+  @override
+  Widget build(BuildContext context) {
+    return Hover(
+      builder: (_) {
+        return Container(
+          height: 40,
+          decoration: BoxDecoration(
+            borderRadius: BorderRadius.circular(6),
+            color: context.color.askButtonBackground,
```
Comment (vovaklh) [inline lib/presentation/pages/directions_page/sections/profit_section.dart:64]: Create new style or move this button to widgets

File: lib/presentation/pages/directions_page/sections/profit_section.dart [hunk 15]
```diff
@@ -0,0 +1,115 @@
...
+
+  @override
+  Widget build(BuildContext context) {
+    return Hover(
+      builder: (_) {
+        return Container(
+          height: 40,
+          decoration: BoxDecoration(
+            borderRadius: BorderRadius.circular(6),
+            border: Border.all(color: context.color.partnerButtonBorder),
```
Comment (vovaklh) [inline lib/presentation/pages/directions_page/sections/profit_section.dart:98]: Same as above

File: lib/presentation/themes/src/constants.dart [hunk 16]
```diff
@@ -27,16 +27,10 @@ abstract class AppColors {
   static const blueGrey = Color(0xFFCACBDF);
   static const darkerBlueGrey = Color(0xFF9596AF);
 
-  static const purple = Color(0xFF7967D6);
-  static const darkPurple = Color(0xFF6055A9);
-  static const lightPurple = Color(0xFF9A8DE9);
-  static const lighterPurple = Color(0xFFB2A4FF);
-  static const lightestPurple = Color(0xFFD6CEFF);
-  static const whitePurple = Color(0xFFF5F2FC);
```
Comment (vovaklh) [inline lib/presentation/themes/src/constants.dart:35]: Why deleted?

File: lib/presentation/themes/src/theme_text_styles.dart [hunk 17]
```diff
@@ -98,6 +98,13 @@ class ThemeTextStyles extends ThemeExtension<ThemeTextStyles> {
   final TextStyle revenueSectionNewsCardText;
   final TextStyle revenueSectionRateCard;
   final TextStyle revenueSectionRateCardHighlighted;
+  final TextStyle partnershipSubtitle;
+  final TextStyle partnershipTitle;
+  final TextStyle directionCardTitle;
+  final TextStyle directionCardDescription;
+  final TextStyle directionsBenefitsTitle;
+  final TextStyle directionsBenefitsDescription;
+  final TextStyle directionsBenefitsSubtitle;
```
Comment (vovaklh) [inline lib/presentation/themes/src/theme_text_styles.dart:107]: pls add "section" to the name. Remember check copyWith() method after renaming

File: lib/presentation/themes/src/theme_text_styles.dart [hunk 18]
```diff
@@ -1045,6 +1093,39 @@ class ThemeTextStyles extends ThemeExtension<ThemeTextStyles> {
...
+          letterSpacing: .3,
+        ),
+        directionsBenefitsTitle: headline1.copyWith(
+          fontSize: 53,
+          color: AppColors.darkBlue,
+        ),
+        directionsBenefitsSubtitle: headline3.copyWith(
+          fontSize: 16,
+          height: 1.12,
+          color: AppColors.indigo,
```
Comment (vovaklh) [inline lib/presentation/themes/src/theme_text_styles.dart:1127]: Do we need height parameter?

File: pubspec.yaml [hunk 19]
```diff
@@ -50,7 +50,8 @@ dependencies:
   locale_emoji_flutter: ^0.2.1
   carousel_slider: ^4.1.1
   auto_size_text: ^3.0.0
-
+  flutter_staggered_grid_view: ^0.6.2
```
Comment (vovaklh) [inline pubspec.yaml:53]: Pls move to UI section

**Review by vovaklh — CHANGES_REQUESTED** [review]

File: lib/presentation/themes/src/constants.dart (hunk 16 above)
Comment (vitalii-petrun) [reply lib/presentation/themes/src/constants.dart:35]: purple isn't needed anymore, Aurora UI is builded via indigo

File: lib/presentation/themes/src/theme_text_styles.dart (hunk 18 above)
Comment (vitalii-petrun) [reply lib/presentation/themes/src/theme_text_styles.dart:1127]: I just copied styles from figma, I think it's needed, cause it's not everywhere in styles, usually it's used in multiline text

File: lib/presentation/pages/directions_page/directions_page.dart (hunk 3 above)
Comment (vitalii-petrun) [reply lib/presentation/pages/directions_page/directions_page.dart:51]: seems like we dont need it, every widget normally has padding

File: lib/presentation/pages/directions_page/directions_page.dart (hunk 2 above)
Comment (vovaklh) [reply lib/presentation/pages/directions_page/directions_page.dart:49]: Replace vertical padding by spaces

File: lib/presentation/pages/directions_page/directions_page.dart (hunk 1 above)
Comment (vitalii-petrun) [reply lib/presentation/pages/directions_page/directions_page.dart:36]: gradient image is not cover all the area, also scaffold is not pure white, so it's ug;y to see

so it's needed

File: lib/presentation/themes/src/theme_text_styles.dart (hunk 18 above)
Comment (vovaklh) [reply lib/presentation/themes/src/theme_text_styles.dart:1127]: Pls check visual diffrence with this height parameter and without. If this is no difference, pls delete all

File: lib/presentation/pages/directions_page/directions_page.dart (hunk 1 above)
Comment (vovaklh) [reply lib/presentation/pages/directions_page/directions_page.dart:36]: Ok, move to styles

**Review by vovaklh — APPROVED** [review]

**Pull request merged**